core/src/main/java/org/apache/iceberg/StaticTableOperations.java

```diff
@@ -34,11 +34,24 @@ public class StaticTableOperations implements TableOperations {
 
   /** Creates a StaticTableOperations tied to a specific static version of the TableMetadata */
   public StaticTableOperations(String metadataFileLocation, FileIO io) {
-    this(metadataFileLocation, io, null);
+    this(null, metadataFileLocation, io, null);
+  }
+
+  public StaticTableOperations(TableMetadata staticMetadata, FileIO io) {
+    this(staticMetadata, staticMetadata.metadataFileLocation(), io, null);
   }
 
   public StaticTableOperations(
       String metadataFileLocation, FileIO io, LocationProvider locationProvider) {
+    this(null, metadataFileLocation, io, locationProvider);
+  }
+
+  public StaticTableOperations(
+      TableMetadata staticMetadata,
+      String metadataFileLocation,
```

Comment:
I don't think we want to expose both `staticMetadata` and `metadataFileLocation`

Reply:
This diff should be preferred IMO:

```
--- a/core/src/main/java/org/apache/iceberg/StaticTableOperations.java
+++ b/core/src/main/java/org/apache/iceberg/StaticTableOperations.java
@@ -44,6 +44,18 @@ public class StaticTableOperations implements TableOperations {
     this.locationProvider = locationProvider;
   }
 
+  public StaticTableOperations(TableMetadata staticMetadata, FileIO io) {
+    this(staticMetadata, io, null);
+  }
+
+  public StaticTableOperations(
+      TableMetadata staticMetadata, FileIO io, LocationProvider locationProvider) {
+    this.staticMetadata = staticMetadata;
+    this.metadataFileLocation = staticMetadata.metadataFileLocation();
+    this.io = io;
+    this.locationProvider = locationProvider;
+  }
```

Reply:
done